pkg/gen/nodejs-templates/yaml.ts.mustache

```diff
@@ -304,39 +315,25 @@ import * as outputs from "../types/output";
                 name = `${config.resourcePrefix}-${name}`
             }
             super("kubernetes:yaml:ConfigFile", name, config, opts);
-            const fileId = config && config.file || name;
-            let text: Promise<string>;
-            if (isUrl(fileId)) {
-                text = fetch(fileId).then(r => {
-                    if (r.ok) {
-                        return r.text()
-                    } else {
-                        throw Error(`Error fetching YAML file '${fileId}': ${r.status} ${r.statusText}`);
-                    }
-                });
-            } else {
-                text = Promise.resolve(fs.readFileSync(fileId).toString());
-            }
+            const path = config && config.file || name;
+
+            const promise = pulumi.output(pulumi.runtime.invoke(
+                "kubernetes:yaml:parse",
+                {input: [{name: name, type: "path", value: path}]},
+                {async: true}));
 
-            this.resources = pulumi.output(text.then(t => {
-                try {
-                    return parseYamlDocument({
-                        objs: yamlLoadAll(t),
+            this.resources = promise.apply(p => {
+                return parseYamlDocument({
+                        objs: p.result[name][path],
                         transformations: config && config.transformations || [],
-                        resourcePrefix: config && config.resourcePrefix || undefined
-                    }, {parent: this})
-                } catch (e) {
-                    throw Error(`Error fetching YAML file '${fileId}': ${e}`);
-                }
-            }));
+                        resourcePrefix: config && config.resourcePrefix,
+                    },
+                    pulumi.mergeOptions(opts, {parent: this}));
+            });
         }
     }
 
-    /** @ignore */ function isUrl(s: string): boolean {
-        return s.startsWith("http://") || s.startsWith("https://")
-    }
-
-    /** @ignore */ function parseYamlDocument(
+    /** @ignore */ export function parseYamlDocument(
```

Comment:
I don't seem to follow why we need `parseYamlDocument` and `parseYamlObject` here and why it's used.

Isn't this functionality supposed to be piped through the invoke call now? Why do we need this in TS if we've pushed it into the provider?

Reply:
I may be able to pull a bit more of the logic into the invoked provider code, but the transformation logic and resource creation has to live in the SDK. I should be able to move the list expansion code and hopefully clean this up a little more.